packages/server/src/sdk/app/tables/external/index.ts

```diff
@@ -231,7 +232,10 @@ export async function save(
   // remove the rename prop
   delete tableToSave._rename
 
-  datasource.entities[tableToSave.name] = tableToSave
+  datasource.entities = {
+    ...datasource.entities,
+    ...tables,
+  }
```

Comment:
How come this is required?

Reply:
throughout this function it builds up a list of updated tables in the `tables` map, based on other tables getting updated when relationships are added. These weren't always getting stored correctly.